### ISSUE TITLE
Not all characters supported in toplevel attribute name

### ISSUE THREAD

**Issue by ptrckb** [INRIA/tralics#17]
The character set that is recognized in the `DocAttrib` command for an attribute name does not match the XML specification (https://www.w3.org/TR/2008/REC-xml-20081126/#d0e804).
For example it does not recognize the ':' character (typically in reserved XML attributes for namespaces).

The argument of `DocAttrib` is splitted with `Buffer::skip_letter()`: https://github.com/tralics/tralics/blob/307104b23be1fe5c620df256cf604e0e14c9d670/src/txtitlepage.cpp#L228

**Comment by vbeffara** [comment]
Yes this whole business of `Buffer` which is a `std::string` with two pointers and the parsing is done by moving the pointers by hand is something that would benefit from refactoring, it makes such changes harder to implement than they should be. This plus the code on lines 241 to 246 in the same file that re-use the current buffer for no apparent reason just to concatenate two temporary strings.

There are plenty of methods (`Buffer::find_top_atts` here, `Buffer::see_equals` that it uses for instance) that I believe would be much clearer if they used regexps ...
